manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -282,6 +289,39 @@ func (tr *TaskReaper) tick() {
 			return nil
 		})
 	}
+
+	// Perform service cleanup
+	deleteServices := make(map[string]struct{})
+	// figure out which services need to be cleaned up
+	if len(tr.cleanupServices) > 0 {
+		tr.store.View(func(tx store.ReadTx) {
+			for svc := range tr.cleanupServices {
+				tasks, err := store.FindTasks(tx, store.ByServiceID(svc))
+				// if all tasks of svc have been cleaned up already, then we should get a response
+				// with length zero. If an error occurs though, we log it and don't mark the service
+				// for removal yet.
+				if err != nil {
+					log.G(context.Background()).WithError(err).Errorf("error in task reaper while looking up tasks for service %s", svc)
+					continue
+				}
+				if len(tasks) == 0 {
+					deleteServices[svc] = struct{}{}
+					delete(tr.cleanupServices, svc)
```

Comment:
Instead of removing one by one, just clear out the map at the end.

Reply:
I do this because I want to keep `cleanupServices` around. It won't necessarily be empty after this loop is run.

Reply:
Should this delete happen later, after the batch service delete?  i.e. After the batch is successful, delete all the services in `deleteServices` from `tr.cleanupServices`.

If the batch fails, then `tr.cleanupServices` will no longer have a reference to the service, and the service may not be updated again so would it be possible that the service never gets cleaned up?